From 1be46de6174640d9f5d7ac9fcce5805cd8ba476e Mon Sep 17 00:00:00 2001
From: sinavir <sinavir@sinavir.fr>
Date: Fri, 27 Jun 2025 21:27:57 +0200
Subject: credentials: Use loadcredentials

---
 manage.py          |  2 +-
 zproject/config.py | 27 ++++++++++++++++++++-------
 2 files changed, 21 insertions(+), 8 deletions(-)

diff --git a/manage.py b/manage.py
index 4d626d0889..11641d1d14 100755
--- a/manage.py
+++ b/manage.py
@@ -126,7 +126,7 @@ if __name__ == "__main__":
     config_file = configparser.RawConfigParser()
     config_file.read("/etc/zulip/zulip.conf")
     PRODUCTION = config_file.has_option("machine", "deploy_type")
-    HAS_SECRETS = os.access("/etc/zulip/zulip-secrets.conf", os.R_OK)
+    HAS_SECRETS = True
 
     if PRODUCTION and not HAS_SECRETS:
         # The best way to detect running manage.py as another user in
diff --git a/zproject/config.py b/zproject/config.py
index 1f44f8660f..42f174d427 100644
--- a/zproject/config.py
+++ b/zproject/config.py
@@ -1,6 +1,8 @@
 import configparser
 import os
-from typing import overload
+from typing import Any, overload
+
+import loadcredential
 
 from scripts.lib.zulip_tools import get_config as get_config_from_file
 
@@ -17,11 +19,8 @@ config_file.read("/etc/zulip/zulip.conf")
 # Whether this instance of Zulip is running in a production environment.
 PRODUCTION = config_file.has_option("machine", "deploy_type")
 DEVELOPMENT = not PRODUCTION
-secrets_file = configparser.RawConfigParser()
-if PRODUCTION:  # nocoverage
-    secrets_file.read("/etc/zulip/zulip-secrets.conf")
-else:
-    secrets_file.read(os.path.join(DEPLOY_ROOT, "zproject/dev-secrets.conf"))
+
+credentials = loadcredential.Credentials()
 
 
 @overload
@@ -35,7 +34,21 @@ def get_secret(
 ) -> str | None:
     if development_only and PRODUCTION:  # nocoverage
         return default_value
-    return secrets_file.get("secrets", key, fallback=default_value)
+    return credentials.get(key, default=default_value)
+
+
+@overload
+def get_secret_json(
+    key: str, default_value: None = None, development_only: bool = False
+) -> Any: ...
+@overload
+def get_secret_json(key: str, default_value: Any, development_only: bool = False) -> Any: ...
+def get_secret_json(
+    key: str, default_value: Any | None = None, development_only: bool = False
+) -> Any:
+    if development_only and PRODUCTION:  # nocoverage
+        return default_value
+    return credentials.get_json(key, default=default_value)
 
 
 def get_mandatory_secret(key: str) -> str:
